daemon/execdriver/lxc/lxc_init_linux.go

```diff
@@ -4,19 +4,38 @@ package lxc
 
 import (
 	"fmt"
+	"os"
 
 	"github.com/docker/libcontainer/utils"
+	"github.com/vishvananda/netns"
 )
 
 func finalizeNamespace(args *InitArgs) error {
 	if err := utils.CloseExecFrom(3); err != nil {
 		return err
 	}
+
+	// Make sure you setup the netns before you setup user since once you
+	// setup as a unprivileged used you cannot set into different namespace.
+	if args.NetNsPath != "" {
```

Comment:
Have you tested this with LXC driver after dropping NET_ADMIN? LXC Caps are set in LXC(so by this time, they are already in place)